src/services/item/plugins/shortLink/repository.ts

```diff
@@ -52,57 +72,42 @@ export const ShortLinkRepository = AppDataSource.getRepository(ShortLink).extend
         createdAt: 'ASC',
       },
     });
+  }
 
-    return shortLinks;
-  },
-
-  async get(alias: string): Promise<ShortLink> {
-    if (!alias) throw new ShortLinkNotFound(alias);
-
-    const shortLink = await this.findOne({
-      where: { alias },
+  async getOne(alias: string): Promise<ShortLink> {
+    const shortLink = await super.findOne(alias, {
       relations: {
         item: true,
       },
     });
 
-    if (!shortLink) throw new ShortLinkNotFound(alias);
+    if (!shortLink) {
+      throw new ShortLinkNotFound(alias);
+    }
 
     return shortLink;
-  },
+  }
 
-  async getWithoutJoin(alias: string): Promise<ShortLink> {
-    if (!alias) throw new ShortLinkNotFound(alias);
-
-    const shortLink = await this.findOne({
-      where: { alias },
+  async getOneFlat(alias: string): Promise<ShortLink> {
+    const shortLink = await super.findOne(alias, {
       select: ShortLink.getAllColumns(AppDataSource.manager),
     });
 
-    if (!shortLink) throw new ShortLinkNotFound(alias);
-
-    return shortLink;
-  },
+    if (!shortLink) {
+      throw new ShortLinkNotFound(alias);
+    }
 
-  async deleteOne(alias: string): Promise<ShortLink> {
-    const shortLink = await this.get(alias);
-    await this.delete(alias);
     return shortLink;
-  },
+  }
 
-  async updateOne(alias: string, postLink: ShortLinkPatchPayload): Promise<ShortLink> {
+  async updateOne(alias: string, entity: UpdateShortLinkBody): Promise<ShortLink> {
     try {
-      if (!alias) throw new ShortLinkNotFound(alias);
-
-      await this.update(alias, postLink);
-      const aliasId = postLink['alias'] ?? alias;
-      return await this.get(aliasId); // return the updated short link
+      return await super.updateOne(alias, entity);
     } catch (e) {
-      // if a duplication entry error throw, the postlink must contain the alias
-      if (isDuplicateEntryError(e)) {
-        throw new ShortLinkDuplication(postLink['alias']);
+      if (isDuplicateEntryError(e.message)) {
```

Comment:
Isn't `isDuplicatedEntryError` waiting to receive an error instead of a string?

Reply:
Yes it does, but apparently, because of InsertionException is encapsulating the error in message property I have to fetch it that way (So I don't break Frontend).

It works because `error` type is `any`

Reply:
Ok I better understand what you proposed with the nested exception.. What do you think about 1) handling the DuplicatedError in the insert method of the Repository and throw a EntityDuplicatedError in it and 2) using the exception's message in the generic InsertionException instead of the error (I made a mistake here, I should have to use the message error)?